src/Shared/src/CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -153,24 +163,40 @@ public int Execute(params string[] args)
 
                         if (option == null)
                         {
-                            if (string.IsNullOrEmpty(longOptionName) && !command._throwOnUnexpectedArg  && AllowArgumentSeparator)
+                            var ignoreContinueAfterUnexpectedArg = false;
+                            if (string.IsNullOrEmpty(longOptionName) &&
+                                !command._throwOnUnexpectedArg &&
+                                AllowArgumentSeparator)
                             {
-                                // skip over the '--' argument separator
+                                // Skip over the '--' argument separator then consume all remaining arguments. All
+                                // remaining arguments are unconditionally stored for further use.
                                 index++;
+                                ignoreContinueAfterUnexpectedArg = true;
```

Comment:
Could we change this to say 

```
command.RemainingArguments.AddRange(new ArraySegment<string>(args, index, args.Length - index));
```

it's much easier to follow and you don't need to introduce a new parameter to this method.

Reply:
Considered that but duplicating the code factored into `HandleUnexpectedArg(...)` would make maintenance more difficult, requires another `break` and wasn't as clear.